### ISSUE TITLE
CPU cap (flag -cpu) with a small percentage is silently ignored

### ISSUE THREAD

**Issue by detaoin** [mholt/caddy#1962]
### 1. What version of Caddy are you using (`caddy -version`)?
Caddy 0.10.10 (non-commercial use only)
However it is also present on current master (commit 1125a236eabb61bbccb5b6a1af1a48e39da59a20)

### 2. What are you trying to do?
N/A (just reading the source code)

### 3. What is your entire Caddyfile?
_empty_

### 4. How did you run Caddy (give the full command and describe the execution environment)?
GODEBUG=schedtrace=1000 caddy -cpu 1%

### 5. Please paste any relevant HTTP request(s) here.
N/A

### 6. What did you expect to see?
The _console_ (stderr) output should contain the scheduler state summary line every second:
```
SCHED 0ms: gomaxprocs=4 idleprocs=1 threads=5 spinningthreads=1 idlethreads=0 runqueue=0 [0 0 0 0]
Activating privacy features... done.
http://localhost:1515
SCHED 1009ms: gomaxprocs=1 idleprocs=1 threads=8 spinningthreads=0 idlethreads=3 runqueue=0 [0]
```

The `gomaxprocs=X` information changes from 4 to 1 after the command line flags have been parsed; we requested (`-cpu 1%`) a very small cpu cap, thus logically _saturate_ at 1 GOMAXPROC, the minimum possible value. (*Note* on this machine I have 4 logical CPUs).

### 7. What did you see instead (give full error messages and/or log)?
```
SCHED 0ms: gomaxprocs=4 idleprocs=1 threads=4 spinningthreads=1 idlethreads=0 runqueue=0 [0 0 0 0]
Activating privacy features... done.
http://localhost:1515
SCHED 1009ms: gomaxprocs=4 idleprocs=4 threads=7 spinningthreads=0 idlethreads=2 runqueue=0 [0 0 0 0]
```

At https://github.com/mholt/caddy/blob/97710ced7ea5b5d8025dc6a3e43d82e9a280920e/caddy/caddymain/run.go#L238 the available number of logical CPUs (`availCPU`) is multiplied by the `1%` specified cap, however the result is rounded down to the nearest integer (0), and ignored by https://golang.org/pkg/runtime/#GOMAXPROCS because smaller than 1.

`numCPU` should have a lower cap of 1, this can be implemented with

```
if numCPU < 1 {
        numCPU = 1
}
```

After line 238, or 250. (inside `if strings.HasSuffix(cpu, "%") {` or after)

### 8. How can someone who is starting from scratch reproduce the bug as minimally as possible?
Run `GODEBUG=schedtrace=1000 caddy -cpu 1%` with any/empty Caddyfile. However the issue can't be reproduced if the machine has more than 99 logical CPUs.

**Comment by mholt** [comment]
Would you submit a pull request? It should be easy to add a test and implement your change. :+1:

**Comment by detaoin** [comment]
I submitted a pull request. However there are two approaches I see to solving the question:

1. Exceptionally round the GOMAXPROCS value up in case of small percent values.
2. Return an error

The argument I would see for number 1, is that for all other valid flag values (percent >= 100/maxCPU), the actual CPU cap is rounded down. Are there situations where a user would like to specify a _hard_ CPU cap?

Personally, I would prefer approach 1 (the one implemented by the PR): the downside of having the server not start (startup error) because I set a CPU cap flag to a value below an obscure threshold which needs knowledge of the number of logical CPUs on the machine to be defined is quite annoying.
